### PR TITLE
Create hyperlinks to other ASF projects for PRJ-XXXX style cross-issue links.

### LINKED ISSUE #129: Should we create links to Jira issues other than Lucene?
For example, there are cross-issue links such as `SOLR-XXXX`, `INFRA-XXXX`, or `LEGAL-XXXX`. At least we can/should create links to frequent projects in Lucene?

### PR DESCRIPTION
**Pull request by mocobeta**
Close #129.

This captures all Jira issue keys like `{PRJ}-{NUMBER}` in issue descriptions/comments and generates hyperlinks to the issues if the `{PRJ}` is known ASF projects.

### CODE REVIEW

**Comment by mocobeta** [comment]
For example, this contains lots of links to SOLR.
https://github.com/mocobeta/migration-test-3/issues/560

And this contains a few links to INFRA.
https://github.com/mocobeta/migration-test-3/issues/559

**Comment by mikemccand** [comment]
Hi, did you test on a few issues?  Hard to parse these regexps!  But the change looks good to me.

I wonder if any of these projects that we may link to have also migrated to GitHub issues?  But to link to the right issue we would need their mapping, but I think this will be rare for us, so let's not worry about it.

**Review by mikemccand — APPROVED** [review]

**Comment by mocobeta** [comment]
I think it'd be difficult to ensure the coverage. There might be false negatives since this regex is rather strict. At least we can log the detected links and check them.

**Comment by mocobeta** [comment]
I run this on ~8000 issues; here are the generated issue links (sampled).
```
https://issues.apache.org/jira/browse/SOLR-15
https://issues.apache.org/jira/browse/SOLR-81
https://issues.apache.org/jira/browse/SOLR-81
https://issues.apache.org/jira/browse/SOLR-81
https://issues.apache.org/jira/browse/SOLR-81
https://issues.apache.org/jira/browse/SOLR-81
https://issues.apache.org/jira/browse/SOLR-261
https://issues.apache.org/jira/browse/SOLR-3251
https://issues.apache.org/jira/browse/SOLR-2189
https://issues.apache.org/jira/browse/SOLR-2189

https://issues.apache.org/jira/browse/PYLUCENE-26
https://issues.apache.org/jira/browse/INFRA-1197
https://issues.apache.org/jira/browse/TIKA-369
https://issues.apache.org/jira/browse/MAHOUT-694
https://issues.apache.org/jira/browse/MAHOUT-19
https://issues.apache.org/jira/browse/HADOOP-3164
https://issues.apache.org/jira/browse/INFRA-3389
```

Looks fine to me.
